llvm/include/llvm/Analysis/TargetTransformInfo.h

```diff
@@ -1788,6 +1788,11 @@ class TargetTransformInfo {
 
   /// @}
 
+  /// For \p F, call \p Body with the name and value of each launch bound.
+  void forEachLaunchBound(
```

Comment:
This is a very specific, not general concept. The name and comment needs to elaborate for what the context is.

Could this be a query to return the set of attributes and values, rather than the callback?

Reply:
> This is a very specific, not general concept. The name and comment needs to elaborate for what the context is.

@jdoerfert Do you have any suggestion about the right way to document this?  Are there any additional attributes you would like for GCNTTIImpl::collectLaunchBounds or NVPTXTTIImpl::collectLaunchBounds to collect?

> Could this be a query to return the set of attributes and values, rather than the callback?

Done.

Reply:
The naming discussion continues at: https://github.com/llvm/llvm-project/pull/102944#discussion_r1795842757